packages/expo-modules-core/ios/Arguments/AnyArgument.swift

```diff
@@ -33,14 +33,24 @@ extension Double: AnyArgument {}
 extension CGFloat: AnyArgument {}
 
 extension String: AnyArgument {}
-extension Dictionary: AnyArgument {}
+extension Data: AnyArgument {
+  public static func getDynamicType() -> AnyDynamicType {
+    return DynamicFoundationDataType()
+  }
+}
 
 extension Optional: AnyArgument where Wrapped: AnyArgument {
   public static func getDynamicType() -> AnyDynamicType {
     return DynamicOptionalType(wrappedType: ~Wrapped.self)
   }
 }
 
+extension Dictionary: AnyArgument {
```

Comment:
What do you think about adding a constraint here?

```suggestion
extension Dictionary: AnyArgument where Key: Hashable {
```